pyiron_continuum/fenics/wrappers.py

```diff
@@ -0,0 +1,333 @@
+        self.storage.condition = condition
+
+    @property
+    def value(self):
+        return self.storage.value
+
+    @property
+    def condition(self):
+        return self.storage.condition
+
+    def __call__(self, function_space):
+        return FEN.DirichletBC(function_space, self.value(), self.condition())
+
+    def __str__(self):
+        return str(self.value) + ", " + str(self.condition)
+
+
+class Mesh(FenicsWrapper):
+    """
+    Creates a spatial mesh.
+
+    The input string should include all fenics code to be evaluated, including the class to be called, while all kwargs
+    are exec'd beforehand.
+    """
+    @property
+    def _parser(self):
+        return StringInputParser(known_elements=[
+            Point,
+            BoxMesh, RectangleMesh, UnitSquareMesh, UnitCubeMesh, UnitDiscMesh, UnitTriangleMesh, UnitIntervalMesh,
+            generate_mesh, Tetrahedron, Box, Rectangle, Circle
+        ])
+
+    def _generate(self):
+        for k, v in self.kwargs.items():
+            exec(f'{k} = {v}')
+        mesh = eval(self.input_string)
+        if not isinstance(mesh, FenicsMesh.Mesh):
+            raise TypeError(f'Expected the mesh to be of type dolfin.cpp.mesh.Mesh, but got {type(mesh)}. You may '
+                            f'need to wrap your input string in "generate_mesh()" to convert a mshr mesh to a dolphin '
+                            f'mesh.')
+        return mesh
+
+    # TODO: Make a bunch of helper functions.
+    #       I'm really not happy about the fact it's all string encoded, but don't see a way around it atm -Liam
+    def Circle(self, point, radius, resolution):
+        self.set(f'generate_mesh(Circle(Point({point}), {radius}), {resolution})')
+
+
+class Solver:
+    def __init__(self, job, func_space_class=FEN.FunctionSpace):
+        self._job = job
+        self._V = func_space_class(
+            job.mesh, job.input.element_type, job.input.element_order
+        )  # finite element volume space
+        if job.input.element_order > 1:
+            self._V_g = FEN.VectorFunctionSpace(
+                job.mesh, job.input.element_type, job.input.element_order - 1
+            )
+        else:
+            self._V_g = FEN.VectorFunctionSpace(
+                job.mesh, job.input.element_type, job.input.element_order
+            )
+        self._u = FEN.TrialFunction(self._V)  # u is the unkown function
+        self._v = FEN.TestFunction(self._V)  # the test function
+        self._lhs = None
+        self._rhs = None
+        self._solution = FEN.Function(self._V)
+        self.time_dependent_expressions = []
+        self.assigned_u = None
+
+    @property
+    def lhs(self):
+        if self._lhs is None:
+            self._lhs = self._job.input.lhs(self)
+        return self._lhs
+
+    @property
+    def rhs(self):
+        if self._rhs is None:
+            self._rhs = self._job.input.rhs(self)
+        return self._rhs
+
+    @property
+    def V(self):
```

Comment:
I saw above that this is actually volume? I guess here and below it would be nice to have a one line description (e.g. just "volume") for each property, if it physically makes sense.

Reply:
I can try this, but here I'll make no promises -- it's all canonical (or at least fenics-canonical) FEM symbols and my domain knowledge for FEM is not 100%

Reply:
Added docs to `wrappers.Solver`